File: client/src/pages/EmployeeTimeCardPage/model/types/EmployeeTimeCard.ts
export type WorkingState = {
	id: string;
	day: number;
	state: string | number;
	holiday?: boolean;
};

export interface EmployeeTimeCard {
	id: string;
	employeeFirstName: string;
	employeeLastName: string;
	month: string;
	availableWorkingHoursPerMonth?: number;
	workingStates: WorkingState[];
}

export interface EmployeeTimeCardSchema {
	data?: EmployeeTimeCard[];
	form?: EmployeeTimeCard[];
	selectedMonth: string;
	readonly: boolean;
	isLoading: boolean;
	error?: string;
}

File: Domain/Entities/EmployeeTimeCard.cs
namespace Domain.Entities
{
    public class EmployeeTimeCard
    {
        public Guid Id { get; set; }
        public Guid EmployeeId { get; set; }
        public Employee Employee { get; set; }
        public string Month { get; set; }
        public int AvailableWorkingHoursPerMonth { get; set; }
        public ICollection<WorkingState> WorkingStates { get; set; } = new List<WorkingState>();
    }
}

